holder.yml

```diff
@@ -56,60 +56,90 @@ paths:
     post:
-      summary: Notifies a holder of an available presentation.
-      operationId: notifyPresentationAvailable
+      summary: Initiates an exchange of information.
+      operationId: initiateExchange
       description:
-        This API is the first part of a holder to holder credential exchange over http.
-        Notifies this server-holder that the client-holder (caller) is prepared to deliver a presentation compliant with the provided Verifiable Presentation Request query. This is the first part of a presentation exchange over this HTTP API. This server-holder MUST return the required 'domain' and 'challenge' parameters for the Presentation. This server-holder MAY reply with an alternate Verifiable Presentation Request query if it requires something different than the one proposed by the client-holder. The client-holder MUST return a Presentation in accordance with the response parameters.
+        A client can use this endpoint to initiate an exchange of a particular
+        type. The client can include HTTP POST information related to the
+        details of exchange it would like to initiate. If the server understands
+        the request, it returns a Verifiable Presentation Request. A request
+        that the server cannot understand results in an error.
+      parameters:
+        - $ref: "./components/parameters/path/ExchangeId.yml"
       requestBody:
-        description: Details the client provides to the server to help it decide if the presentation should be made.
+        description:
+          Information related to the type of exchange the client would like
+          to start.
         content:
           application/json:
             schema:
-              $ref: "#/components/schemas/NotifyPresentationAvailableRequest"
+              anyOf:
+                -
+                  {
+                    "type": "object",
+                    "description": "Data necessary to initiate the exchange."
+                  }
+                -
+                  $ref: "#/components/schemas/NotifyPresentationAvailableRequest"
       responses:
         "200":
-          description: Proceed with presentation
+          description: Proceed with exchange.
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/NotifyPresentationAvailableResponse"
+                $ref: "#/components/schemas/VerifiablePresentationRequest"
         "400":
-          description: Request for presentation is malformed
+          description: Request is malformed.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
         "501":
-          description: Not implemented
+          description: Service not implemented.
         "500":
-          description: internal error
-  /presentations/submissions:
+          description: Internal server error.
+  /exchanges/{exchange-id}/{transaction-uuid}:
```

Comment:
So some clarification:
`/exchanges/{exchange-id}/{transaction-uuid}` is got for `PUT` because you already know the UUID of the object you are referrencing - on `POST` you are creating and object, in this case of a certain type, e.g. `/exchanges/{exchange-id}/` makes sense.

In the case of REST we should have the IDs for objects in both URI and in `body`

@msporny does that make sense

Reply:
This article on the topic may help clarify what I mean [https://restfulapi.net/rest-put-vs-post/](https://restfulapi.net/rest-put-vs-post/)

Reply:
following normal collection style naming convention, I would assume:

```
POST /exchanges -> create {id: 123}
PUT or PATCH /exchanges/{id} -> update an existing resource with the given id.
```

I don't know what `transaction-uuid` is... and I suggest we not create it.

Reply:
> @msporny does that make sense

Yes, I think what you want is for `POST /exchanges/{exchange-id}/{transaction-uuid}` to be changed to `PUT  /exchanges/{exchange-id}/{transaction-uuid}` ... if so, I'm good with that.

However, it seems like @mprorock and @OR13 disagree? Still hard for me to tell.

To be clear, Digital Bazaar would like to be able to identify different types of exchanges at different endpoints (load balancing via URL instead of having to look into each message -- useful at scale). We would also like to be able to refer to individual instances of each exchange (again, load balancing via URL instead of having to look into each message -- also useful at scale). Remember that the age verification system does >40 million individual interactions per day, and the number of HTTP calls are multiples of that... so, not having to look into the payload body is of significant importance at scale.

Moving these things into the payload body doesn't allow us to use some of the faster load balancing frameworks out there that only have to look at the first few lines of the HTTP Headers instead of having to consume the entire message.